test/e2e/karmadactl_test.go

```diff
@@ -1154,6 +1154,148 @@ var _ = ginkgo.Describe("Karmadactl options testing", func() {
+					framework.WaitClusterFitWith(controlPlaneClient, newClusterName, func(cluster *clusterv1alpha1.Cluster) bool {
+						for _, taint := range cluster.Spec.Taints {
+							if taint.Key == taintKey && taint.Value == taintProductionValue && taint.Effect == effect {
+								return false
+							}
+						}
+						return true
+					})
+				})
+
+				ginkgo.By(fmt.Sprintf("Applying %s taint to the cluster", productionTaint), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", newClusterName, productionTaint)
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s tainted", newClusterName))).Should(gomega.BeTrue())
+					framework.WaitClusterFitWith(controlPlaneClient, newClusterName, func(cluster *clusterv1alpha1.Cluster) bool {
+						for _, taint := range cluster.Spec.Taints {
+							if taint.Key == taintKey && taint.Value == taintProductionValue && taint.Effect == effect {
+								return true
+							}
+						}
+						return false
+					})
+				})
+
+				ginkgo.By(fmt.Sprintf("Overwriting %s taint with %s taint without --overwrite flag", productionTaint, testTaint), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", newClusterName, testTaint)
+					_, err := cmd.ExecOrDie()
+					gomega.Expect(err).Should(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(err.Error(), fmt.Sprintf("cluster %s already has environment taint(s) with same effect(s) and --overwrite is false", newClusterName))).Should(gomega.BeTrue())
+					framework.WaitClusterFitWith(controlPlaneClient, newClusterName, func(cluster *clusterv1alpha1.Cluster) bool {
+						for _, taint := range cluster.Spec.Taints {
+							if taint.Key == taintKey && taint.Value == taintProductionValue && taint.Effect == effect {
+								return true
+							}
+						}
+						return false
+					})
+				})
+
+				ginkgo.By(fmt.Sprintf("Overwriting %s taint with %s taint using --overwrite flag", productionTaint, testTaint), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", newClusterName, testTaint, "--overwrite")
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s modified", newClusterName))).Should(gomega.BeTrue())
+					framework.WaitClusterFitWith(controlPlaneClient, newClusterName, func(cluster *clusterv1alpha1.Cluster) bool {
+						for _, taint := range cluster.Spec.Taints {
+							if taint.Key == taintKey && taint.Value == taintTestValue && taint.Effect == effect {
+								return true
+							}
+						}
+						return false
+					})
+				})
```

Comment:
While we don't need to ensure that taints are cleaned up even after a test case fails, we still need to test the cleanup capabilities of taints within our test cases.